src/ocaml/preprocess/lexer_ident.mll

```diff
@@ -77,6 +77,8 @@ rule token = parse
       { OPTLABEL label }
   | "?" (lowercase_latin1 identchar_latin1 *) as label ':'
       { OPTLABEL label }
+  | ("let" (symbolchar +)) as op { LETOP op }
+  | ("and" (symbolchar +)) as op { ANDOP op }
```

Comment:
This looks slightly incorrect. In the compiler's lexer, the rule is:
```
| "let" kwdopchar dotsymbolchar * as op
```
with
```
let kwdopchar = ['$' '&' '*' '+' '-' '/' '<' '=' '>' '@' '^' '|']
let dotsymbolchar = ['!' '$' '%' '&' '*' '+' '-' '/' ':' '=' '>' '?' '@' '^' '|']
```

But here `symbolchar` is bigger than `kwdopchar`:
```
['!' '$' '%' '&' '*' '+' '-' '.' '/' ':' '<' '=' '>' '?' '@' '^' '|' '~']
```

So this rule matches more operators than allowed.

Reply:
Right, it was an over-approximation. Now it should be precise.